### DISCUSSION TITLE
Decrease reliance on external E2 extensions

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2714] [Suggestions]
We need to decrease reliance on external extensions, to minimize pains caused by compiler / extension api changes and just have a more feature complete E2.

Sorted by subscribers.

[plycore](https://steamcommunity.com/sharedfiles/filedetails/?id=216044582) - 40k subs
~~[damagecore](https://steamcommunity.com/sharedfiles/filedetails/?id=217370580) - 30k subs~~ #2675
~~[weaponcore](https://steamcommunity.com/sharedfiles/filedetails/?id=452197127) - 21k subs~~ #2683
[sentcore](https://steamcommunity.com/sharedfiles/filedetails/?id=726355111) - 12k subs

[entextracore](https://steamcommunity.com/sharedfiles/filedetails/?id=714598720) - 12k subs
[spritecore](https://steamcommunity.com/sharedfiles/filedetails/?id=811333794) - 12k subs
[beamcore](https://steamcommunity.com/sharedfiles/filedetails/?id=392679625) - 12k subs
[admincore](https://steamcommunity.com/sharedfiles/filedetails/?id=261302649) - 9k subs
[wirespawn](https://steamcommunity.com/sharedfiles/filedetails/?id=442725317) - 8k subs
[bonecore](https://steamcommunity.com/sharedfiles/filedetails/?id=890734320) - 7k subs
[halocore](https://steamcommunity.com/sharedfiles/filedetails/?id=167160200) - 7k subs
[vguicore](https://steamcommunity.com/sharedfiles/filedetails/?id=1381355814) - 6k subs
[shootto](https://steamcommunity.com/sharedfiles/filedetails/?id=168794775) - 6k subs
[serversidefiles](https://steamcommunity.com/sharedfiles/filedetails/?id=220111120) - 5k subs
[antcore](https://steamcommunity.com/sharedfiles/filedetails/?id=827052465) - 5k subs
[webaudio](https://steamcommunity.com/sharedfiles/filedetails/?id=2466875474) - 5k subs
[posecore](https://steamcommunity.com/sharedfiles/filedetails/?id=1192820274) - 4k subs
[atmoscore](https://steamcommunity.com/sharedfiles/filedetails/?id=727599375) - 3k subs
[vehiclecore](https://steamcommunity.com/sharedfiles/filedetails/?id=898511842) - 3k subs
[globalprint](https://steamcommunity.com/sharedfiles/filedetails/?id=710995786) - 3k subs
[stencilcore](https://steamcommunity.com/sharedfiles/filedetails/?id=1898708917) - 2k subs

The most popular ones here should at least be addressed. I split the most significant and general ones that seem realistic to add to base E2.

**Comment by Denneisk** [comment]
Strike bonecore off the list. Posecore is partially implemented. Missing features as follows:
- All flex functions
- Eye functions
- Bone jiggle functions
- Bone scale functions
- Animtime

**Comment by Denneisk** [comment]
Also isn't Atmoscore outside of the scope of Wiremod as it's meant for an external addon?

**Comment by Hazeofdream** [comment]
Can the wire extras e2cores be made into the e2 base? not sure why they are separate. holoanim in particular is extremely useful

**Reply by vurvdev** [reply]
They are separate cause nobody has taken the time to port them.

In order to port them, first need to refactor the code (wire extras code is notoriously bad)

Propcore was originally in wire extras and was ported

**Reply by Hazeofdream** [reply]
holoanim should atleast pass that test as it's pretty basic.

possibly light as well

**Comment by Hazeofdream** [comment]
[particleCore](https://steamcommunity.com/sharedfiles/filedetails/?id=1235042437) and it's variants are also pretty popular. This could be extended upon the existing effects core and is the most plausible one.
